frame/election-provider-support/src/lib.rs

```diff
@@ -674,3 +673,100 @@ pub type BoundedSupportsOf<E> = BoundedSupports<
 
 sp_core::generate_feature_enabled_macro!(runtime_benchmarks_enabled, feature = "runtime-benchmarks", $);
 sp_core::generate_feature_enabled_macro!(runtime_benchmarks_or_fuzz_enabled, any(feature = "runtime-benchmarks", feature = "fuzzing"), $);
+
+/// The limits of an election result. The bounds are defined over the count of element of the
+/// election (voters or targets) or the overall datastructure size.
+///
+/// Ordering: when comparing two instances of `ElectionBounds`, the `count` has priority over the
+/// `size`, ie. if `A.count > B.count`, then `A > B` regardless of their relative `size`.
+#[derive(Clone, Copy, RuntimeDebug, scale_info::TypeInfo, Encode, Decode, Eq)]
+pub struct ElectionBounds {
+	/// The bound on number of elements. `None` means unbounded.
+	pub count: Option<u32>,
+	/// The bound on size, in bytes. `None` means unbounded.
+	pub size: Option<u32>,
+}
```

Comment:
I think we can actually nest two instances of this:

```
struct DataProviderBound { cound, size } 
struct ElectionBounds { voters: DataProviderBound, targets: DataProviderBound } 
```

and have all functions take one argument. The builder patter can hopefully help keep the process of building common patters of this neat.